sdk/iot/provisioning/src/az_iot_provisioning_client_sas.c

```diff
@@ -18,7 +18,7 @@
 #define AMPERSAND '&'
 #define EQUAL_SIGN '='
 #define STRING_NULL_TERMINATOR '\0'
-#define SCOPE_REGISTRATIONS_STRING "%2fregistrations%2f"
+#define SCOPE_REGISTRATIONS_STRING "/registrations/"
```

Comment:
To have a complete implementataion, the `sig`, `se` and `skn` must be encoded using semantics described here: https://docs.microsoft.com/en-us/dotnet/api/system.net.webutility.urlencode?view=netcore-3.1
Also, within the HMAC portion, the target (audience) needs to be encoded as well (this is commented in code but not implemented that way).

(from https://github.com/Azure/azure-iot-sdk-csharp/blob/master/common/src/device/provisioning/transport/ProvisioningSasBuilder.cs#L69)

Reply:
@CIPop are you sure you want the semantics described in S.N.WebUtility.UrlEncode? Or do you want S.T.E.W semantics?
https://docs.microsoft.com/en-us/dotnet/api/system.text.encodings.web.urlencoder?view=netcore-3.1

For example, are you sure ' ' should be encoded as a '+'?

I shared some of the differences in this issue along with an open question of what RFC we should support: What are your thoughts on it?

https://github.com/Azure/azure-sdk-for-c/issues/686

Reply:
This is what's used in the C# SDK that works. I am assuming that code is in-line with the code running on the server (didn't check because we had no complaints about the C# code so far).